src/mame/video/edevices.h

```diff
@@ -0,0 +1,95 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+#ifndef MAME_VIDEO_EDEVICES_H
+#define MAME_VIDEO_EDEVICES_H
+
+#pragma once
+
+
+class edevices_device : public device_t
+{
+public:
+	edevices_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock);
+	edevices_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+	// configuration
+	void set_tags(const char *bgram,const char *mlowram,const char *mhighram,const char *txram,const char *bgscrram,const char *mlowscrram,const char *mhighscrram,const char *attrram,const char *sprram,const char *gfx,const char *pal)
```

Comment:
If you're going to configure tags, you should template it so it can take other tag-likes (usually object finders).

But why does this need to get everything from shares in the host device?  Is it impossible to allocate the memory inside the device itself?  The host devices (in this case drivers) don't seem to access this stuff directly.

Reply:
the memory isn't in the fpga, it's on the board, the fpga accesses the memory on the board, so generally I prefer to do it that way, unless the MAME architecture makes that not work.

Reply:
beyond that, I don't know what you mean, I used the model I could find in 30+ other drivers

Reply:
@cuavas Haze requested you to clarify your code review here

Reply:
Or did you already tackle it in "do stuff modern way (nw)"? You should mark these Resolved then.

Reply:
yes, I changed how that worked in that PR, I thought the 'resolve conversation' was just for the conversation, not the issue.  The issue is resolved in the update, and by that it now shows 'outdated' next to these comments.  I wouldn't have even seen this reply had I not been pointed at it.